telegram/ext/_applicationbuilder.py

```diff
@@ -732,6 +736,27 @@ def arbitrary_callback_data(
         self._arbitrary_callback_data = arbitrary_callback_data
         return self
 
+    def local_mode(self: BuilderType, local_mode: bool) -> BuilderType:
+        """Specifies the value for :paramref:`~telegram.Bot.local_mode` for the
+         :attr:`telegram.ext.Application.bot`.
+         If not called, will default to :obj:`False`.
+
+        .. seealso:: `Local Bot API Server <https://github.com/python-telegram-bot/\
+            python-telegram-bot/wiki/Local-Bot-API-Server>`_,
+
+        Args:
+            local_mode (:obj:`bool`): The setting
+
+        Returns:
+            :class:`ApplicationBuilder`: The same builder with the updated argument.
+        """
+        if self._bot is not DEFAULT_NONE:
+            raise RuntimeError(_TWO_ARGS_REQ.format("local_mode", "bot instance"))
+        if self._updater not in (DEFAULT_NONE, None):
+            raise RuntimeError(_TWO_ARGS_REQ.format("local_mode", "updater"))
```

Comment:
we could perhaps at this point extract this into a method since its repeated so many times